### PR TITLE
feat: SpModal component

### PR DESCRIPTION
**Pull request by Matthews3301**
https://github.com/tendermint/vue/issues/137

### CODE REVIEW

File: packages/vue/src/components/SpModal/SpModal.vue [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    cancelButton: {
+      type: Boolean,
+      defaultsTo: true
+    },
+  },
+  data() {
+    return {
+    }
+  },
+  computed: {
```
Comment (marinhoarthur) [inline packages/vue/src/components/SpModal/SpModal.vue]: can we remove these blanks?

File: packages/vue/src/components/SpModal/SpModal.vue [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+      type: Boolean,
+      defaultsTo: true
+    },
+    cancelButton: {
+      type: Boolean,
+      defaultsTo: true
+    },
+  },
+  data() {
+    return {
```
Comment (marinhoarthur) [inline packages/vue/src/components/SpModal/SpModal.vue]: we should add the emitting of `close` on `esc` press as well

**Review by marinhoarthur — CHANGES_REQUESTED** [review]

File: packages/vue/src/components/SpModal/SpModal.vue (hunk 2 above)
Comment (Matthews3301) [reply packages/vue/src/components/SpModal/SpModal.vue]: Updated

File: packages/vue/src/components/SpModal/SpModal.vue (hunk 1 above)
Comment (Matthews3301) [reply packages/vue/src/components/SpModal/SpModal.vue]: Updated

**Comment by fadeev** [comment]
Looking good. Do you think you can also add this component to Storybook? If you'd like we can tackle this in a separate PR.

**Comment by Matthews3301** [comment]
@fedeev added - using this branch for simplicity

**Review by marinhoarthur — APPROVED** [review]

**Pull request merged**